pypeit/core/extract.py

```diff
@@ -1517,8 +1519,12 @@ def objfind(image, thismask, slit_left, slit_righ, inmask=None, fwhm=3.0, use_us
         plt.xlabel('Approximate Spatial Position (pixels)')
         plt.ylabel('F/sigma (significance)')
         plt.title(qa_title + ': Slit# {:d}'.format(specobj_dict['SLITID']))
-        viewer, ch = display.show_image(image*(thismask*inmask))
-        plt.show()
+        if debug_all:
+            viewer, ch = display.show_image(image*(thismask*inmask))
+            plt.show()
+        elif outfile_objprof is not None:
+            plt.savefig(outfile_objprof, dpi=400)
+            plt.close('all')
```

Comment:
I think the logic here is difficult.  I.e., someone could set `show_peaks=True`, but `debug_all=False` and not provide an `outfile_objprof` and then `pyplot` will have created something that never gets flushed, which will mess up the next plot being made.  It would be good to clean up how the QA is done for this function, but, barring that, I would put the `plt.close('all')` call outside the `if...elif` block.